kramer/src/main/kotlin/kramer/Aars.kt

```diff
@@ -0,0 +1,52 @@
+package kramer
+
+import org.xml.sax.Attributes
+import org.xml.sax.SAXException
+import org.xml.sax.helpers.DefaultHandler
+import java.io.IOException
+import java.net.URI
+import java.nio.file.FileSystemAlreadyExistsException
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import java.nio.file.Path
+import javax.xml.parsers.SAXParserFactory
+
+@Throws(IOException::class)
+internal fun extractPackageFromManifest(fromZip: URI): String? {
+  try {
+    FileSystems.newFileSystem(fromZip, emptyMap<String, Any>())
+  } catch (e: FileSystemAlreadyExistsException) {
+    FileSystems.getFileSystem(fromZip)
+  }
+    .rootDirectories
+    .firstOrNull()
+    ?.let { root: Path ->
+      Files.walk(root)
+        .filter { path -> path.toString() == "/AndroidManifest.xml" }
+        .findFirst()
+        .orElse(null)
+        ?.let { path ->
+          val xmlText = Files.readAllLines(path).joinToString("")
+          val parserFactory = SAXParserFactory.newInstance()
```

Comment:
Companion object? While the factories aren't expensive, they are not exactly cheap.

Reply:
Hah. Oops. Yeah, I meant to extract the factory and forgot.

Reply:
Also, this is in a top-level function, so just a top level private `var` is enough. No companion object needed.